lighthouse-core/gather/driver.js

```diff
@@ -137,6 +137,11 @@ class Driver {
     this.on('Debugger.paused', () => this.sendCommand('Debugger.resume'));
 
     connection.on('protocolevent', this._handleProtocolEvent.bind(this));
+
+    /** @private @deprecated Only available for plugin backcompat. */
```

Comment:
`@private` seemed helpful here to prevent any accidental usage, but it's unclear to me if we consider our JSDoc typedefs part of our public non-breaking API. 

I seem to recall always being cadgy about our hard to use typedefs because we were hesitant to open them up this way, but if I'm wrong there or we'd really like to keep it accessibile I'm open to other ideas to prevent accidental use :)

Reply:
If you made these functions instead of assigning as properties, the `@deprecated` tag would would:

![image](https://user-images.githubusercontent.com/4071474/106186538-0ac45a00-616a-11eb-86ce-fc70e1189b21.png)

To prevent us from using it, we just need to find a linter (or tsc?) rule to error on usage of deprecated functions.